controllers/deployment_test.go

```diff
@@ -256,6 +256,131 @@ var _ = Describe("Controller Suite", func() {
 			}
 		})
 	})
+
+	Describe("Add additional generic container", func() {
+		readMaster := func(fileName string) *v1alpha1.CDAPMaster {
+			master := &v1alpha1.CDAPMaster{}
+			err := fromJson(fileName, master)
+			Expect(err).To(BeNil())
+			return master
+		}
+		readExpectedJson := func(fileName string) []byte {
+			json, err := ioutil.ReadFile("testdata/" + fileName)
+			Expect(err).To(BeNil())
+			return json
+		}
+		diffJson := func(expected, actual []byte) {
+			opts := jsondiff.DefaultConsoleOptions()
+			diff, text := jsondiff.Compare(expected, actual, &opts)
+			Expect(diff.String()).To(Equal(jsondiff.SupersetMatch.String()), text)
+		}
+		diffAdditionalContainers := func(containers []corev1.Container, containerName, expectedJsonFilename string) {
+			var additionalContainer *corev1.Container
+			for _, c := range containers {
+				if c.Name == containerName {
+					additionalContainer = &c
+					break
+				}
+			}
+
+			actualContainerJson, _ := json.Marshal(additionalContainer)
+			expectedContainerJson := readExpectedJson(expectedJsonFilename)
+			diffJson(actualContainerJson, expectedContainerJson)
+		}
+		It("Multiple Additional containers for Router", func() {
```

Comment:
nit: I think this should read like a sentence: "It should add multiple containers fro Router". Similarly for other tests.